### PR TITLE
JoinHostPort method used wherever needed

### PR DESCRIPTION
**Pull request by nisarg1499**
### What problem does this PR solve?

Fix SemGrep complaint to use JoinHostPort instead of Sprintf for network address construction.

```
opt.semgrep.sprintf-host-port
use net.JoinHostPort instead of fmt.Sprintf(c.GRPCPort, c.Host)
```
### What is changed and how does it work?
Replaced fmt.Sprintf with net.JoinHostPort, because fmt.Sprintf will cause issues in IPv6. Best practice is to use net.JoinHostPort
### Checklist

Tests

- [ ] Unit test
- [ ] E2E test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Breaking backward compatibility

Related changes

- [ ] Need to update the documentation

### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: pkg/chaosdaemon/server.go [hunk 1]
```diff
@@ -61,12 +61,12 @@ type tlsConfig struct {
 
 // Get the http address
 func (c *Config) HttpAddr() string {
-	return fmt.Sprintf("%s:%d", c.Host, c.HTTPPort)
+	return net.JoinHostPort(c.Host, fmt.Sprintf("%d", HTTPPort))
```
Comment (STRRL) [inline pkg/chaosdaemon/server.go:64]:
```suggestion
	return net.JoinHostPort(c.Host, fmt.Sprintf("%d", c.HTTPPort))
```

File: pkg/chaosdaemon/server.go [hunk 2]
```diff
@@ -61,12 +61,12 @@ type tlsConfig struct {
...
 }
 
 // Get the grpc address
 func (c *Config) GrpcAddr() string {
-	return fmt.Sprintf("%s:%d", c.Host, c.GRPCPort)
+	return net.JoinHostPort(c.Host, fmt.Sprintf("%d", GRPCPort))
```
Comment (STRRL) [inline pkg/chaosdaemon/server.go:69]: ditto

**Review by STRRL — CHANGES_REQUESTED** [review]
Hi @nisarg1499, thanks for your contribution!

Here are something that needs to fix.

**Comment by STRRL** [comment]
And do not forget to format the codes with `go fmt`, or execute `make check` directly. 😁

File: pkg/chaosdaemon/server.go (hunk 1 above)
Comment (nisarg1499) [reply pkg/chaosdaemon/server.go:64]: Ohh. I'll fix this.

File: pkg/chaosdaemon/server.go (hunk 2 above)
Comment (nisarg1499) [reply pkg/chaosdaemon/server.go:69]: Yaa this too.

File: pkg/chaosdaemon/server.go (hunk 1 above)
Comment (nisarg1499) [reply pkg/chaosdaemon/server.go]: Change done @STRRL

File: pkg/chaosdaemon/server.go (hunk 2 above)
Comment (nisarg1499) [reply pkg/chaosdaemon/server.go]: Change done @STRRL

**Comment by nisarg1499** [comment]
All change are done @STRRL 
Please review it.

**Review by STRRL — APPROVED** [review]
LGTM

**Review by yeya24 — APPROVED** [review]
LGTM

**Comment by nisarg1499** [comment]
@STRRL The jenkins job has succeeded. Then why Tide is showing not mergeable? How to check what's wrong in this case?

**Comment by STRRL** [comment]
> @STRRL The jenkins job has succeeded. Then why Tide is showing not mergeable? How to check what's wrong in this case?

Thanks for your contribution! It requires maintainers to approve to run the CI for first-contribution of contributor, I just start it, and I think it will be merged very soon.

**Comment by nisarg1499** [comment]
@STRRL The e2e-test failed. I wanted to see the job logs in jenkins, but somehow I am not able to access Jenkins link. What should I do? Also any idea what could go wrong?

**Comment by YangKeao** [comment]
> @STRRL The e2e-test failed. I wanted to see the job logs in jenkins, but somehow I am not able to access Jenkins link. What should I do? Also any idea what could go wrong?

:facepalm: It seems to be an internal error. Now the CI cluster has resumed and we will try to merge this PR asap.

**Pull request merged**